storage/service_account/hmac/hmac_test.go

```diff
@@ -122,7 +123,7 @@ func TestGetKey(t *testing.T) {
 	key, err := createTestKey(tc.ProjectID, t)
 	defer deleteTestKey(key)
```

Comment:
AFAICT, this helper doesn't check key being nil before acting upon on it. I understand the error isn't occuring on this line, but just wondering if this defer should occur after the code knows the key creation didn't fail. 

```golang
func deleteTestKey(key *storage.HMACKey) {
	ctx := context.Background()
	handle := storageClient.HMACKeyHandle(key.ProjectID, key.AccessID)
	if key.State == "ACTIVE" {
		handle.Update(ctx, storage.HMACKeyAttrsToUpdate{State: "INACTIVE"})
	}
	if key.State != "DELETED" {
		handle.Delete(ctx)
	}
}
```

Reply:
That's a good point-- I actually think it'd be better to add a nil check to the delete function as well. I will do that.